src/ring_theory/dedekind_domain.lean

```diff
@@ -0,0 +1,58 @@
+import ring_theory.fractional_ideal
+import ring_theory.ideal.over
+import ring_theory.discrete_valuation_ring
+import order.zorn
+
+namespace ring
+
+/-- A ring `R` is at most one-dimensional if all nonzero prime ideals are maximal. -/
+def dimension_le_one (R : Type*) [comm_ring R] : Prop :=
+∀ p ≠ (⊥ : ideal R), p.is_prime → p.is_maximal
+
+end ring
+
+open ideal ring
+
+lemma is_principal_ideal_ring.dimension_le_one (R : Type*) [integral_domain R]
+  [is_principal_ideal_ring R] :
+  ring.dimension_le_one R :=
+λ p nonzero prime, by { haveI := prime, exact is_prime.to_maximal_ideal nonzero }
+
+variables {R K : Type*}
+
+lemma integral_closure.dimension_le_one [comm_ring R] [nontrivial R] [integral_domain K] [algebra R K]
+  (h : dimension_le_one R) :
+  dimension_le_one (integral_closure R K) :=
+begin
+  intros p ne_bot prime,
+  haveI := prime,
+  refine integral_closure.is_maximal_of_is_maximal_comap p (h _ (integral_closure.comap_ne_bot ne_bot) _),
+  apply is_prime.comap
+end
+
+/-- A Dedekind domain is a nonfield is Noetherian, integrally closed, and has Krull dimension exactly one.
```

Comment:
Do you want to mention something about the fact that the comment says "exactly one" and the code "`le_one`"?

Reply:
Yeah, it's a bit odd. I added that it's implied by it being a nonfield and `le_one`.